### PR TITLE
Praat pitch tracking in audioGUI

### PR DESCRIPTION
**Pull request by cwnaber**
### Summary of changes:
wave_viewer, and by extension audioGUI, now support pitch tracking through Praat. When pitch is calculated, if the sigproc_params.ptrack_method field is set to 'praat', a Praat system call will be executed.
* wave_viewer\make_pitch_axdat now calls get_sig_pitch with new parameters, namely, passing sigproc_params directly.
* get_sig_pitch now switches based on sigproc_params.ptrack_method being 'praat' or otherwise (the legacy functionality). Legacy functionality **has been changed**. Output arguments 3 and 4 have been removed. Output argument 2 is now the pitch time axis `pitch_taxis`
* get_sig_pitch in 'praat' mode executes a system command to Praat based on sigproc_params. This is based on wave_viewer\get_formant_tracks.m
* get_sig_pitch has a new header

* wave_viewer.m\get_sigproc_params() has the new pitch tracking parameters. These are only referenced if the data wasn't passed in via audioGUI. If the signal was passed in through audioGUI, default parameters are set in free-speech/speech/get_sigproc_defaults.m

* new function get_pitch_tracks.praat is the Praat script which is run via system command. It uses the To Pitch (ac)... function. Like with wave_viewer\get_formant_tracks.m, it temporarily saves files temp_wav.wav and temp_wav_pitch.txt in the wave_viewer folder. These are subsequently deleted by get_sig_pitch.m.

### Workflows used for testing:
* **Brand new dataset**. Given only expt.mat and data.mat, open audioGUI, save wave_viewer_params, run waverunner
* **Converting old dataset**. Given an expt.mat, data.mat, wave_viewer_params.mat and trial files created in old format, run waverunner with `waverunner([],[1],[],0,{'pitch', 'pitch_taxis', 'pitchlimits'})`
* **Verify old method still works**. Given an expt.mat and data.mat , open audioGUI with command `audioGUI([], [], [], [], 0, [], 'ptrack_method', 'not-praat')`

### Merge timing
This should be merged simultaneously with [free-speech#120](https://github.com/carrien/free-speech/pull/120).

### CODE REVIEW

**Comment by cwnaber** [comment]
@carrien New changes:
* get_sig_pitch no longer has variable output arguments. It just returns the pitch values vector and `pitch_taxis`
* Calls to get_sig_pitch no longer need an if/else clause. These calls have been updated.

**Comment by cwnaber** [comment]
@carrien Oh, and I did test generating a `dataVals.mat` file. It does successfully make a file and accurately populate `dataVals.f0` and `dataVals.pitch_taxis` even though the pitch time axis is unique.

It seems like the first value of `dataVals.f0` is very frequently NaN. I haven't looked into how exactly `gen_dataVals_from_wave_viewer` works such that that's the case.

File: get_sig_pitch.m
Comment (carrien) [inline get_sig_pitch.m:10]:
```suggestion
% In the Praat method, the `pitchlimits` input argument is ignored, and
```
We can leave things like this now, but when we get to the step where we're changing the GUI, we could consider replacing the 3rd input argument "pitchlimits" with "params" (and then, inside this function, set pitchlimits = params.pitchlimits to be used by the non-praat algorithm). It could be a good opportunity to get rid of the redundancy.

File: get_sig_pitch.m
Comment (carrien) [inline get_sig_pitch.m:69]:
```suggestion
        if ismac
            praatExePath = "/Applications/Praat.app/Contents/MacOS/Praat"
        else
            praatExePath = "C:\Users\Public\Desktop\Praat.exe";
        end
        praatCmd = sprintf('%s --run get_pitch_tracks.praat %s %s %d .... etc. etc. etc.
        status = system(praatCmd);
```
Even if you don't want to use sprintf, it's definitely better to split out the path to praat.exe as a variable, because that appears to be the only thing that differs between the two commands; this way we can just have one command. (But sprintf will save you from all those num2str's and also double-quotes within single-quotes!)

**Review by carrien — APPROVED** [review]
Approving even though I requested a change so it doesn't hold up the merge. (I can approve again when you make the change.) Also see comment about future changes when we update the GUI.

File: get_sig_pitch.m
Comment (parrell) [reply get_sig_pitch.m:69]: Yes, I cleaned this up later this afternoon, at least to get rid of the redundancy though not using  sprint.

Comment (parrell) [reply get_sig_pitch.m:69]: yes i cleaned this up this afternoon, though not using sprintf

**Pull request merged**
